examples/llm/src/models/utils/make_llama_work_with_autoclass.py

```diff
@@ -0,0 +1,329 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""LLaMA model configuration and tokenizer.
+
+REMOVE THIS CODE when we upgrade to a version of transformers that has the
+LlamaTokenizer code
+"""
+
+import os
+from shutil import copyfile
+from typing import Any, Dict, List, Optional, Tuple
+
+import sentencepiece as spm
+from transformers import AutoTokenizer
+from transformers.configuration_utils import PretrainedConfig
+from transformers.tokenization_utils import AddedToken, PreTrainedTokenizer
+from transformers.utils import logging
+
+
+class LlamaConfig(PretrainedConfig):
+    model_type = 'llama'
+
+    def __init__(
+        self,
+        vocab_size=32000,
+        hidden_size=4096,
+        intermediate_size=11008,
+        num_hidden_layers=32,
+        num_attention_heads=32,
+        hidden_act='silu',
+        max_position_embeddings=2048,
+        initializer_range=0.02,
+        rms_norm_eps=1e-6,
+        use_cache=True,
+        pad_token_id=0,
+        bos_token_id=1,
+        eos_token_id=2,
+        tie_word_embeddings=False,
+        **kwargs,
+    ):
+        self.vocab_size = vocab_size
+        self.max_position_embeddings = max_position_embeddings
+        self.hidden_size = hidden_size
+        self.intermediate_size = intermediate_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.hidden_act = hidden_act
+        self.initializer_range = initializer_range
+        self.rms_norm_eps = rms_norm_eps
+        self.use_cache = use_cache
+        super().__init__(
+            pad_token_id=pad_token_id,
+            bos_token_id=bos_token_id,
+            eos_token_id=eos_token_id,
+            tie_word_embeddings=tie_word_embeddings,
+            **kwargs,
+        )
+
+
+class mC4SentencePieceConfig(LlamaConfig):
+
+    def __init__(self, *args, **kwargs):
+        super.__init__(
+            *args,
+            vocab_size=65500,
+            pad_token_id=3,
+            tie_word_embeddings=False,
+            **kwargs,
+        )
+
+
+##
+# tokenizer
+##
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {'vocab_file': 'tokenizer.model'}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    'vocab_file': {
+        'hf-internal-testing/llama-tokenizer':
+            'https://huggingface.co/hf-internal-testing/llama-tokenizer/resolve/main/tokenizer.model',
+    },
+    'tokenizer_file': {
+        'hf-internal-testing/llama-tokenizer':
+            'https://huggingface.co/hf-internal-testing/llama-tokenizer/resolve/main/tokenizer_config.json',
+    },
+}
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    'hf-internal-testing/llama-tokenizer': 2048,
+}
+
+
+class LlamaTokenizer(PreTrainedTokenizer):
```

Comment:
no fast?

Reply:
No, we need a newer version of `tokenizers` to do that, and we were worried about changing deps